### PR TITLE
[pkg/translator/prometheus] fix panic for duplicated feature gate registration in otel-collector and prometheus packages

### PR DESCRIPTION
**Pull request by gracewehner**
**Description:**
This mitigates the bug that causes the otelcollector to crash when a project depends on both opentelemetry-collector-contrib and prometheus packages. This is a fix until the code is maintained in a single package and MustRegister() could be used again.

The variable here is initialized by a function that attempts registration. If that errors, search the global registry for a gate that already exist with that ID. If none is still found, then panic.

The same changes will need to be copied over to the duplicate files in the prometheus repository so that both are not calling MustRegister().

**Link to tracking Issue:**
https://github.com/prometheus/prometheus/issues/13842

**Testing:**
Added tests that explicitly re-register the flag to ensure there's no panic and the existing gate returned is expected.

### CODE REVIEW

File: pkg/translator/prometheus/normalize_label.go [hunk 1]
```diff
@@ -4,19 +4,46 @@
...
 
-var dropSanitizationGate = featuregate.GlobalRegistry().MustRegister(
+var dropSanitizationGate = registerOrLoadGate(
 	"pkg.translator.prometheus.PermissiveLabelSanitization",
 	featuregate.StageAlpha,
 	featuregate.WithRegisterDescription("Controls whether to change labels starting with '_' to 'key_'."),
 	featuregate.WithRegisterReferenceURL("https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/8950"),
 )
 
+func registerOrLoadGate(id string, stage featuregate.Stage, opts ...featuregate.RegisterOption) *featuregate.Gate {
```
Comment (dashpole) [inline pkg/translator/prometheus/normalize_label.go:21]:
```suggestion
// registerOrLoadGate is a temporary workaround for a featuregate collision.
// Remove this after https://github.com/open-telemetry/opentelemetry-collector/issues/9859 is fixed.
func registerOrLoadGate(id string, stage featuregate.Stage, opts ...featuregate.RegisterOption) *featuregate.Gate {
```

Comment (dashpole) [reply pkg/translator/prometheus/normalize_label.go:21]: also, consider naming this "mustRegisterOrLoadGate", since it panics.

Comment (gracewehner) [reply pkg/translator/prometheus/normalize_label.go]: Thanks @dashpole! Added these changes

**Review by dashpole — APPROVED** [review]

**Review by andrzej-stencel — APPROVED** [review]
Looks good to me

**Comment by Aneurysm9** [comment]
I don't think this is the way we should go with this.  The feature gate registrations [will be removed](https://github.com/prometheus/prometheus/pull/13932) from Prometheus and they're [planning a hard fork](https://github.com/prometheus/prometheus/pull/13991) so there's little likelihood of the conflict being reintroduced.  I don't believe there's any urgent need to work around this here at this time.

**Pull request closed without merging**